docs/our-approach-to-data.md

```diff
@@ -0,0 +1,301 @@
+
+- Module directories in `lib`
+- Modules include `actions.js` and at least one store (named or suffixed `store.js`)
+- Action creators dispatch view or server actions on the global `Dispatcher` object
+- Stores include a top-level object for data storage, which is not directly exported
+- Stores export a number of helper getter functions for accessing known data
+- Stores subscribe to the Dispatcher, manipulating data in response to action types it is concerned with
+
+__Advantages:__
+
+- Stores can be specialized to their specific needs since dispatched actions are run against all subscribing stores
+- There is a single entry point by which data can enter the store. This is easier to manage as an application scales
+- Data logic (e.g. fetching) is not intertwined with the storage of the data
+- Adopting an accepted pattern grants us access to a community-driven ecosystem of reference implementations
+
+### Third Era: Redux Global State Tree (December 2015 - Present)
+
+[Redux](http://redux.js.org/), described as a "predictable state container", is an evolution of the principles advocated in Flux. It is not a far departure from Flux, but is unique in many ways:
+
+- There is typically a single store instance which maintains all state for the entire application
+- Action creators do not call to the global dispatcher directly, but rather return simple action objects which can be passed to the [store `dispatch` method](http://rackt.org/redux/docs/api/Store.html#dispatch)
+- While Flux Stores are responsible for maintaining own state, Redux reducers are composable functions that manipulate specific parts of the global state "tree"
+- Since state is the [single source of truth](http://rackt.org/redux/docs/introduction/ThreePrinciples.html#single-source-of-truth) for the entire application, reducers tend to be much simpler and transparent than Flux stores
+
+__Identifying characteristics:__
+
+- Files exist within the `state` directory, mirroring the structure of the global tree
+- React bindings use [`react-redux`](https://github.com/rackt/react-redux) `connect`
+
+__Advantages:__
+
+- An arguably simpler abstraction to the same problems addressed by Facebook’s Flux implementation
+- Better suited for server-side rendering, as the singleton nature of Flux stores exposes the risk of leaking session data between requests
+- Encourages and often forces a developer toward writing functional, testable code
+- Extendable, supporting middlewares to suit our specific needs and [conveniences for use with React](https://github.com/rackt/react-redux)
+
+## Current Recommendations
+
+All new data requirements should be implemented as part of the global Redux state tree. The `client/state` directory contains all of the behavior describing the global application state. The folder structure of the `state` directory should directly mirror the sub-trees within the global state tree. Each sub-tree can include their own reducer, actions, and selectors.
```

Comment:
I really want to avoid the fragmentation of `/shared` | `/client`. Can we figure out a plan for it?